src/js/views/programs/shared/components/toggle_component.js

```diff
@@ -0,0 +1,30 @@
+import hbs from 'handlebars-inline-precompile';
+import { View } from 'marionette';
+
+export default View.extend({
```

Comment:
I think it might be better to make this view something that can re-render on it's own... which means ideally the toggle is in the template and not the className  This also really isn't a component.  could make it one... where the state of the component is stored with it, but this could just be a shared view.    Although I'm not sure it even needs to be, https://github.com/RoundingWell/care-ops-frontend/blob/develop/src/js/views/programs/sidebar/program/programs-sidebar_views.js#L92 is pretty much the required functionality if the view is passed the model, so some duplication might be ok
But if you go with a shared view it could be configurable via an `attr` defined like: `const FlowPublishedToggle = ToggleComponent.extend({ attr: 'published_at' });`
```js
export default View.extend({
  template: ProgramStateTemplate,
  triggers: {
    'click': 'click',
  },
  modelEvents: () {
     return {
         [`change:${ this.getOption('attr') }`]: 'render',
     };
  },
  templateContext() {
    status: this.model.get(this.getOption('attr'));
  },
});
```

if this change is made, there wouldn't be need for the parent view to rerender on the change, with exception of if it is disabled.

Reply:
Ah ya, good call. I had three instances of that `ToggleView` created that were almost identical. So I felt compelled to do something to stop the duplication.

I'll play around with the options you mentioned. And probably go with whichever one is simpler in the end. I'm guessing that'll be just the view that re-renders on it's own.